### ISSUE TITLE
Support should not send coverage unless plugins task has been registered

### ISSUE THREAD

**Issue by bahmutov** [cypress-io/code-coverage#179]
in a situation where a support file has been registered BUT there is no task in the plugins file, we should not try to call the coverage report task

Example situation
```json
{
  "pluginsFile": false,
  "supportFile": "node_modules/cypress-react-unit-test/support"
}
```

**Comment by bahmutov** [comment]
<img width="720" alt="Screen Shot 2020-04-06 at 9 08 19 AM" src="https://user-images.githubusercontent.com/2212006/78561691-2b3fb400-77e6-11ea-9a14-6a25be14e22c.png">

Maybe we should use `window.__coverage__` to know?

**Comment by bahmutov** [comment]
Maybe if plugins can set a variable when registering a task we could do
```js
// user code
module.exports = (on, config) => {
  require('@cypress/code-coverage/task')(on, config)
}
```

inside the task 
```js
module.exports = (on, config) => {
  on('task', tasks)
  config.env.codeCoverageTasksRegistered = true
  return config
}
```

**Comment by CharlesStover** [comment]
Is it possible for this to throw an error? I've been debugging for days why I'm not getting any coverage files in my CI workflow. Local tests didn't give me anything either, and this warning was simply not visible:
![image](https://user-images.githubusercontent.com/343837/147002035-81519988-eb3d-4da2-85b6-ae0af737dcc7.png)
It was sheer happenstance that I ran a one-liner test on a local Cypress instance and saw this banner.

The end problem was that my `plugins` file's default export wasn't returning `config` after I mutated it with `@cypress/code-coverage`:

```
import task from '@cypress/code-coverage';
export default function(on, config) {
  task(on, config);
  return config; // <-- REQUIRED
}
```

This would have been so much easier to debug if CI had failed with an error that the code coverage task had not registered instead of failing silently.

**Comment by taymoork2** [comment]
For those who run into the edge case of trying to add `process.env` to `config.env` (i.e. `config.env = process.env`)
This comment from issue #405 helped resolved the issue of the code coverage task not registering
 
https://github.com/cypress-io/code-coverage/issues/405#issuecomment-882820159

tldr:
```js
require('@cypress/code-coverage/task')(on, config)
config.env = {
    ...process.env,
    ...config.env, // Don't overwrite `codeCoverageTasksRegistered` set by `@cypress/code-coverage/task`
  };
```

**Comment by gwvt** [comment]
How do you keep the coverage tasks registered when using multiple config files as documented at https://docs.cypress.io/api/plugins/configuration-api#Customize-available-browsers?

In `plugins/index.js` as documented there:
```
module.exports = (on, config) => {
  const environment = config.env.configFile || "development";
  const configurationForEnvironment = fetchConfigurationByFile(environment);

  return configurationForEnvironment || config;
};
```
How do you incorporate
```
  require('@cypress/code-coverage/task')(on, config);
```
to register the coverage tasks on the returned config object? Config by file works but without coverage. Coverage works without config by file (i.e., just returning `config`). I can't get both to work together...
